providers/implementations/asymciphers/rsa_enc.c

```diff
@@ -162,10 +162,12 @@ static int rsa_encrypt(void *vprsactx, unsigned char *out, size_t *outlen,
         if ((tbuf = OPENSSL_malloc(rsasize)) == NULL)
             return 0;
         if (prsactx->oaep_md == NULL) {
-            OPENSSL_free(tbuf);
             prsactx->oaep_md = EVP_MD_fetch(prsactx->libctx, "SHA-1", NULL);
-            ERR_raise(ERR_LIB_PROV, ERR_R_INTERNAL_ERROR);
-            return 0;
+            if (prsactx->oaep_md == NULL) {
```

Comment:
Not sure why we got the encrypt wrong but the decrypt right here..
This fallback is still kind of flawed since it doesnt use propq.. 
A sensible app should be setting this via the digest param.